powerauth-restful-security-javaee/src/main/java/io/getlime/security/powerauth/rest/api/jaxrs/provider/PowerAuthAuthenticationProvider.java

```diff
@@ -243,8 +243,9 @@ public PowerAuthApiAuthentication validateRequestSignature(
         PowerAuthApiAuthentication auth;
         try {
             auth = this.authenticate(powerAuthAuthentication);
-        } catch (RemoteException e) {
-            throw new PowerAuthAuthenticationException("POWER_AUTH_SIGNATURE_SOAP_ERROR");
+        } catch (RemoteException ex) {
+            logger.warn("Remote communication failed, error: {}", ex.getMessage());
+            throw new PowerAuthAuthenticationException("POWER_AUTH_SIGNATURE_REMOTE_ERROR");
```

Comment:
Let's pass `ex` as a parameter to the new exception to keep the cause somewhere.